File: Volume 10 (1000 - 1099)/1045/1045.rb
#!/usr/bin/env ruby

t = (Array.new(gets.split.map(&:to_f))).sort
if t[2] >= t[1] + t[0]
    puts "NAO FORMA TRIANGULO"
else
    if t[2]**2 == t[1]**2 + t[0]**2
        puts "TRIANGULO RETANGULO"
    elsif t[2]**2 > t[1]**2 + t[0]**2
        puts "TRIANGULO OBTUSANGULO"
    elsif t[2]**2 < t[1]**2 + t[0]**2
        puts "TRIANGULO ACUTANGULO"
    end
    if t[0] == t[1] and t[1] == t[2]
        puts "TRIANGULO EQUILATERO"
    elsif t[0] == t[1] or t[1] == t[2] or t[0] == t[2]
        puts "TRIANGULO ISOSCELES"
    end
end

File: Volume 10 (1000 - 1099)/1045/1045.c
#include <stdio.h>
 
int main() {
    double a, b, c, inter;
    scanf("%lf %lf %lf", &a, &b, &c);
    if (b > a && b > c) {
        inter = a;
        a = b;
        b = inter;
    }
    if (c > a && c > b) {
        inter = a;
        a = c;
        c = inter;
    }
    if (a >= b + c)
        printf("NAO FORMA TRIANGULO\n");
    else {
        if (a*a == b*b + c*c)
            printf("TRIANGULO RETANGULO\n");
        else if (a*a > b*b + c*c)
            printf("TRIANGULO OBTUSANGULO\n");
        else if (a*a < b*b + c*c)
            printf("TRIANGULO ACUTANGULO\n");
        if (a == b && b == c)
            printf("TRIANGULO EQUILATERO\n");
        else if (a == b || b == c || a == c)
            printf("TRIANGULO ISOSCELES\n");
    }
    return 0;
}